pkgs/_test_macros/lib/query_class.dart

```diff
@@ -18,7 +20,14 @@ class QueryClassImplementation implements Macro {
 
   @override
   Future<AugmentResponse> augment(Host host, AugmentRequest request) async {
```

Comment:
I think in the future we will want a `Future<void>` for the return type here, and there should be methods on `host` to perform various augmentations.

This will make macros more composable without needing to merge augmentation responses together.

The host probably needs to be phase specific as well. We might also want phase specific response types. And then phase specific methods that you implement with those more specific types on the parameters. Maybe `augmentTypes`, `augmentDeclarations`, etc.

Reply:
That all makes sense, thanks.